components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/AddRoleWFRequestHandler.java

```diff
@@ -220,6 +220,10 @@ public boolean isValidOperation(Entity[] entities) throws WorkflowException {
                         workflowService.entityHasPendingWorkflowsOfType(entities[i], UserStoreWFConstants
                                 .UPDATE_ROLE_NAME_EVENT))) {
                     throw new WorkflowException("Rolename already exists in the system. Please pick another rolename.");
+                } else if (workflowService.eventEngagedWithWorkflows(UserStoreWFConstants.ADD_USER_EVENT) &&
+                        entities[i].getEntityType() == UserStoreWFConstants.ENTITY_TYPE_USER && workflowService
```

Comment:
Isn't that we have to use 'equals()' instead '==' as we are doing a object content comparison?